rust/psibase/src/services/tokens/precision.rs

```diff
@@ -4,39 +4,56 @@ use fracpack::{Pack, ToSchema, Unpack};
 
 use crate::{serialize_as_str, services::tokens::ConversionError};
 
-#[derive(PartialEq, Debug, Copy, Clone, Pack, Unpack, ToSchema)]
+#[derive(PartialEq, Debug, Copy, Clone, Pack, ToSchema)]
 #[fracpack(fracpack_mod = "fracpack")]
-pub struct Precision {
-    pub value: u8,
-}
+pub struct Precision(pub u8);
 
 impl TryFrom<u8> for Precision {
     type Error = ConversionError;
 
     fn try_from(value: u8) -> Result<Self, Self::Error> {
         if value <= 8 {
-            Ok(Self { value })
+            Ok(Self(value))
         } else {
             Err(ConversionError::PrecisionOverflow)
         }
     }
 }
 
+impl<'a> Unpack<'a> for Precision {
+    const FIXED_SIZE: u32 = 1;
+
+    const VARIABLE_SIZE: bool = false;
+
+    fn unpack(src: &'a [u8], pos: &mut u32) -> fracpack::Result<Self> {
+        Ok(Self(u8::unpack(src, pos)?))
```

Comment:
This should also check. e.g. `...try_into().map_err(|_| fracpack::Error::BadScalar)`